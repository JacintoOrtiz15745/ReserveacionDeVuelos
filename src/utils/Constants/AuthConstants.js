const GoogleId = '555459120478-5dvb5sj1o0996s86gg8bk69ogm7vdiu4.apps.googleusercontent.com';  
export const AuthConstants = {
  GoogleId,
};
