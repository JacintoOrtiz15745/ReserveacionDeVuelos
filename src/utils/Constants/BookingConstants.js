const TitleOrigin = 'Where are you now?';
const TitleDestination = 'Where will you be flying to?';
const TextInput = 'Select location';
const BtnText = 'Next';
const SelectDate = 'Select Date';
const Passengers = 'How many passengers?';
const BookingToRoute = 'BookingTo';
const BookingNowRoute = 'BookingNow';
const MyFlightsRoute = 'MyFlights';
const BookingDateRoute = 'BookingDate';
const BookingDoneRoute = 'BookingDone';
const BookingPassengersRoute = 'BookingPassenger';
const doneMessage = 'Your request was received.';
const finish = 'Finish';
const PickerData = [
  {
    label: 'Amsterdam, Netherlands',
    value: 'AMS,Netherlands',
  },
  {
    label: 'Berlin, Germany',
    value: 'BER,Germany',
  },
  {
    label: 'Paris, France',
    value: 'PAR,France',
  },
  {
    label: 'London, UK',
    value: 'LON,UK',
  },
  {
    label: 'Madrid, Spain',
    value: 'MAD,Spain',
  },
];
const passengerNumberData = [
  {
    id: '1',
    title: '1',
  },
  {
    id: '2',
    title: '2',
  },
  {
    id: '3',
    title: '3',
  },
  {
    id: '4',
    title: '4',
  },
  {
    id: '5',
    title: '5',
  },
];
export const BookingConstants = {
  doneMessage,
  TitleOrigin,
  TitleDestination,
  TextInput,
  BtnText,
  SelectDate,
  Passengers,
  BookingToRoute,
  BookingNowRoute,
  MyFlightsRoute,
  passengerNumberData,
  PickerData,
  BookingDateRoute,
  BookingPassengersRoute,
  BookingDoneRoute,
  finish,
};
