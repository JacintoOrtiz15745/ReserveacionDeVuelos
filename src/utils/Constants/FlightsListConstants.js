const planeIconName = 'plane';
const passengers = 'passengers';
const myFlightsTitle = 'My flights';
const flightDataExperimental = [
  {
    id: '1',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'Serbia',
    destinyCity: 'Netherlands',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '2',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '3',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '4',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '5',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '6',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
  {
    id: '7',
    originCountry: 'BEG',
    destinyCountry: 'AMS',
    originCity: 'dsfsf',
    destinyCity: 'fdgdg',
    date: 'September 3, 2021',
    passengers: 5,
  },
];

export const FlightsListConstants = {
  planeIconName,
  passengers,
  myFlightsTitle,
  flightDataExperimental,
  
};
