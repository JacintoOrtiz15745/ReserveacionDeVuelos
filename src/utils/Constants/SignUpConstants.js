const SignUpTitle = 'Sign Up';
const NameInputTitle = 'First Name';
const EmailInputTitle = 'Email *';
const PasswordInputTitle = 'Password *';
const TermsCheckboxTitle = 'I agree to the Terms and Privacy Policy.';
const UpdatesCheckboxTitle = 'Subscribe for select product updates.';
const CharactersMessage =
  'Use 8 or more characters with a mix of letters, numbers, and simbols.';
const SignUpButton = 'Sign Up';
const GoogleSignUpButton = 'Sign Up with Google';
const HasAnAccount = 'Already have an account?';
const LogIn = 'Log In';
const nameErrorMessage = ' must be at least 3 characters.';
const emailErrorMessage = 'invalid Email format';
const passwordErrorMessage = 'must be at least 8 characters';
const OR = 'or';
const SignIn = 'Sign In';
const SignUp = 'Sign Up';
const noAccountMsg = 'DonÂ´t have an account?';
export const SignUpConstants = {
  SignUpTitle,
  NameInputTitle,
  EmailInputTitle,
  PasswordInputTitle,
  TermsCheckboxTitle,
  UpdatesCheckboxTitle,
  CharactersMessage,
  SignUpButton,
  GoogleSignUpButton,
  HasAnAccount,
  LogIn,
  nameErrorMessage,
  emailErrorMessage,
  passwordErrorMessage,
  OR,
  SignIn,
  noAccountMsg,
  SignUp,
};
