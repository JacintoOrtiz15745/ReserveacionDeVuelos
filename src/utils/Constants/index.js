
export * from './SignUpConstants';
export * from './FlightsListConstants'
export * from './AuthConstants'
export * from './BookingConstants'