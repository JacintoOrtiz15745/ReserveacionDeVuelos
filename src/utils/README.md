This folder will contain functions and utilities if needed.