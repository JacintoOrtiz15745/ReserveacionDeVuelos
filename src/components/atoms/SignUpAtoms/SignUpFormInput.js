import React from 'react';
import {Text, TextInput, View} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import {SignUpConstants} from '../../../utils/Constants';
import {colors} from '../../../styles/Colors';
const SignUpFormInput = (props) => {
  const {
    inputName,
    value,
    validation,
    errorMessage,
    setFormValues,
    handleChange,
    formValues,
    isValidated,
  } = props;

  return (
    <View>
      <Text style={styles.formText} testID="inputTitle">
        {inputName}
        {!validation && <Text style={styles.errorMessage}>{errorMessage}</Text>}
      </Text>
      <TextInput
        testID="input"
        value={value}
        style={[
          styles.textInput,
          {borderColor: !isValidated ? colors.black : colors.Purple},
        ]}
        onChangeText={(text) => handleChange(formValues, setFormValues, text)}
        secureTextEntry={
          inputName === SignUpConstants.PasswordInputTitle ? true : false
        }
      />
      {inputName === SignUpConstants.PasswordInputTitle && (
        <Text style={styles.characters} testID="passwordMsg">
          {SignUpConstants.CharactersMessage}
        </Text>
      )}
    </View>
  );
};

export default SignUpFormInput;
