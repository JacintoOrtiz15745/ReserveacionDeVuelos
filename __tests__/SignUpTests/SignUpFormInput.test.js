import React from 'react';
import SignUpFormInput from '../../src/components/atoms/SignUpAtoms/SignUpFormInput';
import {shallow} from 'enzyme';
import {SignUpConstants} from '../../src/utils/Constants';

let wrapper;
const formValues = {
  name: '',
  email: '',
  password: '',
  termsCheckbox: false,
  updatesCheckbox: false,
  validName: true,
  validEmail: true,
  validPassword: true,
  validatedName: false,
  validatedEmail: false,
  validatedPassword: false,
};
const props = {
  inputName: SignUpConstants.NameInputTitle,
  value: '',
  validation: true,
  errorMessage: SignUpConstants.nameErrorMessage,
  setFormValues: jest.fn(),
  handleChange: jest.fn(),
  formValues,
  isValidated: false,
};
beforeEach(() => {
  wrapper = shallow(<SignUpFormInput {...props} />);
});
describe('renders correclty when inputName is "name"', () => {
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Input Title', () => {
    const InputTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'inputTitle',
    );
    expect(InputTitle.length).toBe(1);
  });
  test('renders Title string correctly', () => {
    const InputTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'inputTitle',
    );

    expect(InputTitle.props().children[0]).toEqual(props.inputName);
  });
  test('password bottom text does not show when inputName is "name"', () => {
    const passwordMsg = wrapper.findWhere(
      (node) => node.prop('testID') === 'passwordMsg',
    );
    expect(passwordMsg.length).toBe(0);
  });
  test('TextInput handle change works as expected', () => {
    const textInput = wrapper.findWhere(
      (node) => node.prop('testID') === 'input',
    );
    textInput.props().onChangeText();
    wrapper.update();
    expect(props.handleChange).toHaveBeenCalled();
  });
});

describe('renders correclty when inputName is "email"', () => {
  const props = {
    inputName: SignUpConstants.EmailInputTitle,
    value: '',
    validation: true,
    errorMessage: SignUpConstants.emailErrorMessage,
    setFormValues: jest.fn(),
    handleChange: jest.fn(),
    formValues,
    isValidated: false,
  };
  const wrapper = shallow(<SignUpFormInput {...props} />);
  test('renders Title string correctly', () => {
    const InputTitle = wrapper.findWhere(
      (node) => node.prop('testID') === 'inputTitle',
    );

    expect(InputTitle.props().children[0]).toEqual(props.inputName);
  });
  test('password bottom text does not show when inputName is "email"', () => {
    const passwordMsg = wrapper.findWhere(
      (node) => node.prop('testID') === 'passwordMsg',
    );
    expect(passwordMsg.exists()).not.toBeTruthy();
  });
});

describe('renders correclty when inputName is "password"', () => {
  const props = {
    inputName: SignUpConstants.PasswordInputTitle,
    value: '',
    validation: true,
    errorMessage: SignUpConstants.passwordErrorMessage,
    setFormValues: jest.fn(),
    handleChange: jest.fn(),
    formValues,
    isValidated: false,
  };
  const wrapper = shallow(<SignUpFormInput {...props} />);
  test('renders Title string correctly', () => {
    const InputName = wrapper.findWhere(
      (node) => node.prop('testID') === 'inputTitle',
    );
    expect(InputName.props().children[0]).toEqual(props.inputName);
  });
  test('render bottom text when inputname is password', () => {
    const passworMsg = wrapper.findWhere(
      (node) => node.prop('testID') === 'passwordMsg',
    );
    expect(passworMsg.exists()).toBeTruthy();
  });

  test('renders Title string correctly in password Message', () => {
    const passworMsg = wrapper.findWhere(
      (node) => node.prop('testID') === 'passwordMsg',
    );

    expect(passworMsg.props().children).toEqual(
      SignUpConstants.CharactersMessage,
    );
  });
});
