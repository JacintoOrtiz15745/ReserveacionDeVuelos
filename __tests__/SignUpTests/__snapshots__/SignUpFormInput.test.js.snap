// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty when inputName is "name" snapshot 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
    testID="inputTitle"
  >
    First Name
  </Text>
  <Component
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 10,
        },
        Object {
          "borderColor": "black",
        },
      ]
    }
    testID="input"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
